spec/client_spec.rb

```diff
@@ -170,6 +170,21 @@
         end.at_least(:once)
         lock_manager.lock(resource_key, ttl)
       end
+
+      it 'accepts retry_delay as proc' do
+        retry_delay = proc { |attempt_number| 200 * attempt_number ** 2 }
+        lock_manager = Redlock::Client.new(Redlock::Client::DEFAULT_REDIS_URLS,
+                                            lock_manager_opts.merge(retry_delay: retry_delay))
+
+        another_lock_info = lock_manager.lock(resource_key, ttl)
+        attempt_number = 1
+        expect(lock_manager).to receive(:sleep) do |sleep|
+          expect(sleep).to satisfy { |value| value < retry_delay.call(attempt_number) + described_class::DEFAULT_RETRY_JITTER }
+          attempt_number += 1
+        end.at_least(:once)
```

Comment:
I find this test (specifically the 4 lines above) to be a bit hard to read. You could

* set `retry_count` to 1 for this test, so you know how many calls you'll get.
* assert that `attempt_number` passed to you proc is 1
* otherwise ignore the attempt number in your proc, just return `1_000_000_000`
* then you also don't need to call your proc again in your `satisfy` block, which I think improves it a lot 👍
* your `satisfy` is nice already, but better would be a `expect(sleep).to be_within(described_class::DEFAULT_RETRY_JITTER).of(1_000_000_000)`
* `at_least(:once)` -> `exactly(:once)`